File: hh-survivor-backend/src/main/java/hhsurvivor/HHSurvivorApplication.java
package hhsurvivor;

import io.dropwizard.Application;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;

public class HHSurvivorApplication extends Application<HHSurvivorConfiguration> {

    public static void main(final String[] args) throws Exception {
        new HHSurvivorApplication().run(args);
    }

    @Override
    public String getName() {
        return "hh-survivor";
    }

    @Override
    public void initialize(final Bootstrap<HHSurvivorConfiguration> bootstrap) {
        // TODO: application initialization
    }

    @Override
    public void run(final HHSurvivorConfiguration configuration,
                    final Environment environment) {
        // TODO: implement application
    }

}

File: backend/src/main/kotlin/org/hhsurvivor/HHSurvivorApplication.kt
package org.hhsurvivor

import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.module.kotlin.KotlinModule
import de.spinscale.dropwizard.jobs.JobsBundle
import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment

class HHSurvivorApplication(): Application<HHSurvivorConfiguration>() {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            HHSurvivorApplication().run(*args)
        }
    }

    override fun initialize(bootstrap: Bootstrap<HHSurvivorConfiguration>) {
        val downloadGamesJob = DownloadGamesJob()
        bootstrap.addBundle(JobsBundle(downloadGamesJob))
        bootstrap.objectMapper.registerModule(KotlinModule())
        bootstrap.objectMapper.enable(SerializationFeature.INDENT_OUTPUT);

    }

    override fun run(configuration: HHSurvivorConfiguration, environment: Environment) {
        environment.jersey().register(DataResource(environment.objectMapper))
    }

}
